storage/innobase/xtrabackup/src/backup_mysql.cc

```diff
@@ -2163,69 +2164,137 @@ backup_cleanup()
 	}
 }
 
-static ib_mutex_t mdl_lock_con_mutex;
 static MYSQL *mdl_con = NULL;
-
+static MYSQL *mdl_fts_con = NULL;
 void
-mdl_lock_init()
+mdl_lock_tables(MYSQL *connection)
 {
-	mutex_create(LATCH_ID_XTRA_DATAFILES_ITER_MUTEX, &mdl_lock_con_mutex);
-
+	msg_ts("Initializing MDL on all current tables.\n");
+	MYSQL_RES *mysql_result = NULL;
+	MYSQL_ROW row;
 	mdl_con = xb_mysql_connect();
-
+	mdl_fts_con = xb_mysql_connect();
```

Comment:
Do we need separate mdl_fts_conn?

Reply:
@satya-bodapati when we try to translate FTS table, we still have a resultset open on mdl_con
```
		mysql_result = xb_mysql_query(mdl_con, "SELECT NAME, SPACE FROM "
									"INFORMATION_SCHEMA.INNODB_SYS_TABLES", true, true);
		while ((row = mysql_fetch_row(mysql_result))) { <-- result set iterator
			if (fil_is_user_tablespace_id(atoi(row[1]))) {
				char *lock_query;
				char full_table_name[MAX_FULL_NAME_LEN + 1];
				innobase_format_name(full_table_name, sizeof(full_table_name), row[0]);
				if (is_fts_index(full_table_name)) { <-- new query is executed while resultset still open
					msg_ts("%s is a Full-Text Index. Translating table name\n",
									full_table_name);
					fts_get_table_name(full_table_name);
				}
 . . .
		}
		mysql_free_result(mysql_result); <-- closing resultset

Reply:
After offline discussion, we decided that there is no point of translating the FTS index to base table to lock the base table since we will eventually pick up the base table from INNODB_SYS_TABLES.
Will remove mdl_fts_con